### PR TITLE
Fix Code convention warnings in Rider.

### PR DESCRIPTION
**Pull request by surfnerd**

### CODE REVIEW

File: UnitySDK/Assets/ML-Agents/Examples/PushBlock/Scripts/PushAgentBasic.cs [hunk 1]
```diff
@@ -84,8 +84,8 @@ public override void CollectObservations()
         {
             var rayDistance = 12f;
 
-            AddVectorObs(m_RayPer.Perceive(rayDistance, m_RayAngles, m_DetectableObjects, 0f, 0f));
-            AddVectorObs(m_RayPer.Perceive(rayDistance, m_RayAngles, m_DetectableObjects, 1.5f, 0f));
```
Comment (chriselion) [inline UnitySDK/Assets/ML-Agents/Examples/PushBlock/Scripts/PushAgentBasic.cs]: This doesn't look right - the args default to 0.0 but the second line is a behavior change

File: UnitySDK/Assets/ML-Agents/Scripts/Policy/BarracudaPolicy.cs [hunk 2]
```diff
@@ -32,7 +32,7 @@ public class BarracudaPolicy : IPolicy
             NNModel model,
             InferenceDevice inferenceDevice)
         {
-            var aca = GameObject.FindObjectOfType<Academy>();
+            var aca = Object.FindObjectOfType<Academy>();
```
Comment (vincentpierre) [inline UnitySDK/Assets/ML-Agents/Scripts/Policy/BarracudaPolicy.cs]: Why this change ?

File: UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/TensorGenerator.cs [hunk 3]
```diff
@@ -36,13 +36,11 @@ public interface IGenerator
...
         /// Returns a new TensorGenerators object.
         /// </summary>
-        /// <param name="bp"> The BrainParameters used to determine what Generators will be
-        /// used</param>
         /// <param name="seed"> The seed the Generators will be initialized with.</param>
         /// <param name="allocator"> Tensor allocator</param>
+        /// <param name="memories">Dictionary of AgentInfo.id to memory for use in the inference model.</param>
         /// <param name="barracudaModel"></param>
         public TensorGenerator(
-            BrainParameters bp,
```
Comment (vincentpierre) [inline UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/TensorGenerator.cs]: Good catch

File: UnitySDK/Assets/ML-Agents/Scripts/Sensor/CameraSensorComponent.cs [hunk 4]
```diff
@@ -9,7 +9,7 @@ public class CameraSensorComponent : SensorComponent
         public string sensorName = "CameraSensor";
         public int width = 84;
         public int height = 84;
-        public bool grayscale = false;
```
Comment (chriselion) [inline UnitySDK/Assets/ML-Agents/Scripts/Sensor/CameraSensorComponent.cs]: I never know which of these suggestions to ignore. For example, it grays out 'private" when it's redundant, but that doesn't seem to be caught here.
https://github.com/Unity-Technologies/ml-agents/blob/58cee7ec0243725c99b573ecc36cf288f7f76975/UnitySDK/Assets/ML-Agents/Scripts/Sensor/CameraSensor.cs#L9-L14

File: UnitySDK/Assets/ML-Agents/Examples/PushBlock/Scripts/PushAgentBasic.cs (hunk 1 above)
Comment (surfnerd) [reply UnitySDK/Assets/ML-Agents/Examples/PushBlock/Scripts/PushAgentBasic.cs]: fat fingered

File: UnitySDK/Assets/ML-Agents/Scripts/Policy/BarracudaPolicy.cs (hunk 2 above)
Comment (surfnerd) [reply UnitySDK/Assets/ML-Agents/Scripts/Policy/BarracudaPolicy.cs]: The static method is actually on Object, not GameObject

File: UnitySDK/Assets/ML-Agents/Scripts/Sensor/CameraSensorComponent.cs (hunk 4 above)
Comment (surfnerd) [reply UnitySDK/Assets/ML-Agents/Scripts/Sensor/CameraSensorComponent.cs]: this was more about the default value being redundant.  `false` is default for `bool`

File: UnitySDK/Assets/ML-Agents/Examples/PushBlock/Scripts/PushAgentBasic.cs (hunk 1 above)
Comment (surfnerd) [reply UnitySDK/Assets/ML-Agents/Examples/PushBlock/Scripts/PushAgentBasic.cs]: fixed
